src/lib/form-field/form-field.md

```diff
@@ -0,0 +1,129 @@
+in RTL), and one end-aligned.
+
+Hint labels are specified in one of two ways: either by using the `hintLabel` property of
+`<mat-form-field>`, or by adding a `<mat-hint>` element inside the form field. When adding a hint
+via the `hintLabel` property it will be treated as the start hint. Hints added via the
+`<mat-hint>` hint element can be added to either side by setting the `align` property on
+`<mat-hint>` to either `start` or `end`. Attempting to add multiple hints to the same side will
+raise an error.
+
+<!-- example(form-field-hint) -->
+
+### Error messages
+Error messages can be shown under the form field underline by adding `mat-error` elements inside the
+form field. Errors are hidden initially and will be displayed on invalid form fields after the user
+has interacted with the element or the parent form has been submitted. Since the errors occupy the
+same space as the hints, the hints are hidden when the errors are shown.
+
+If an form field can have more than one error state, it is up to the consumer to toggle which
+messages should be displayed. This can be done with CSS, `ngIf` or `ngSwitch`. Multiple error
+messages can be shown at the same time if desired, but the `<mat-form-field>` only reserves enough
+space to display one error message at a time. Ensuring that enough space is available to display
+multiple errors is up to the user.
+
+<!-- example(form-field-error) -->
+
+### Prefix & suffix
+HTML can be included before and after the input tag, as a prefix or suffix. It will be underlined
+as per the Material specification.
+
+Adding the `matPrefix` directive to an element inside the `<mat-form-field>` will designate it as
+the prefix. Similarly, adding `matSuffix` will designate it as the suffix.
+
+<!-- example(form-field-prefix-suffix) -->
+
+### Custom form field controls
+In addition to the form field controls that Angular Material provides, it is possible to create
+custom form field controls that work with `<mat-form-field>` in the same way. For additional
+information on this see the guide on
+[Creating Custom mat-form-field Controls](https://material.angular.io/guide/creating-a-custom-form-field-control).
+
+### Theming
+`<mat-form-field>` has a `color` property which can be set to `primary`, `accent`, or `warn`. This
+will set the color ofr the form field underline and floating placeholder based on the theme colors
+of your app.
+
+`<mat-form-field>` inherits its `font-size` from its parent element. This can be overridden to an
+explicit size using CSS. We recommend a specificity of at least 1 element + 1 class.
+
+```css
+mat-form-field.mat-form-field {
+  font-size: 16px;
+}
+```
+
+<!-- example(form-field-theming) -->
+
+### Accessibility
+If a floating placeholder is specified it will be automatically used as the label for the form field
+control. If no floating placeholder is specified the user should label the form field control
```

Comment:
If no floating placeholder is specified the user -> If no floating placeholder is specified, the user